src/components/select-menu/SelectMenu.spec.tsx

```diff
@@ -0,0 +1,220 @@
+        // If it was not present, add it
+        setSelectedOptions([...selectedOptions, option]);
+      else {
+        // else, remove from array of selected options
+        setSelectedOptions(
+          [...selectedOptions].filter(item => item.value !== option.value)
+        );
+      }
+    } else {
+      setSelectedOptions([option]);
+    }
+  };
+
+  return (
+    <SelectMenu
+      options={[
+        {
+          value: 'physics',
+          label: 'Physics',
+        },
+        {value: 'history', label: 'History'},
+        {value: 'science', label: 'Science'},
+      ]}
+      placeholder="Select..."
+      selectedOptions={selectedOptions}
+      onOptionChange={handleOptionChange}
+      disabled={props.disabled}
+      valid={props.valid}
+      invalid={props.invalid}
+      {...props}
+    />
+  );
+}
+
+window.HTMLElement.prototype.scroll = jest.fn();
+
+describe('<SelectMenu />', () => {
+  it('renders SelectMenu in the default state', () => {
+    const select = render(<RenderSelectMenu />);
+    const selectElement = select.getByRole('combobox') as HTMLElement;
+
+    expect(select.getByText('Select...')).toBeTruthy();
+    expect(selectElement.getAttribute('aria-expanded')).toBe('false');
+    expect(selectElement.getAttribute('aria-disabled')).toBeFalsy();
+    expect(selectElement.getAttribute('aria-invalid')).toBeFalsy();
+  });
+
+  it('opens options popup when select element is clicked', async () => {
+    const select = render(<RenderSelectMenu />);
+    const selectElement = select.getByRole('combobox') as HTMLElement;
+
+    userEvent.click(selectElement);
+    expect(selectElement.getAttribute('aria-expanded')).toEqual('true');
+    select.findByRole('listbox');
+    expect(select.getByRole('listbox').getAttribute('id')).toBe(
+      select.getByRole('combobox').getAttribute('aria-controls')
+    );
+    expect(
+      within(select.getByRole('listbox')).getAllByRole('option').length
+    ).toBeTruthy();
+  });
+
+  it('can select single option', async () => {
+    const select = render(<RenderSelectMenu />);
+    const selectElement = select.getByRole('combobox') as HTMLElement;
+
+    userEvent.click(selectElement);
+    await select.findByRole('listbox');
+    fireEvent.click(select.getByText('Physics'));
+
+    await waitFor(() => expect(select.queryByRole('listbox')).toBeFalsy());
+    expect(select.queryByText('Select...')).toBeFalsy();
+    expect(select.getByText('Physics')).toBeTruthy();
+
+    userEvent.click(selectElement);
+    await select.findByRole('listbox');
+    fireEvent.click(select.getByText('History'));
+    await waitFor(() => expect(select.queryByRole('listbox')).toBeFalsy());
+    expect(select.getByText('History')).toBeTruthy();
+
+    userEvent.click(selectElement);
+    await waitFor(() => expect(select.queryByRole('listbox')).toBeTruthy());
+    expect(
+      within(select.getByRole('option', {selected: true})).getByText('History')
```

Comment:
I think this test might be more readable? If we get particular element and compare it to the ohter it's good practise to keep those elements referenced in variables.

e.g
```ts
const option = select.getByRole('option', {selected: true});
const item = within(option).getByText('History');
expect(item).toBeVisible();
```